autoload/lsp/ui/vim.vim

```diff
@@ -487,12 +487,21 @@ function! s:handle_location(ctx, server, type, data) abort "ctx = {counter, list
                 echo 'Retrieved ' . a:type
                 botright copen
             else
-                let l:lines = readfile(fnameescape(l:loc['filename']))
-                call lsp#ui#vim#output#preview(a:server, l:lines, {
-                            \   'statusline': ' LSP Peek ' . a:type,
-                            \   'cursor': { 'line': l:loc['lnum'], 'col': l:loc['col'], 'align': g:lsp_peek_alignment },
-                            \   'filetype': &filetype
-                            \ })
+                if has_key(l:loc,'viewstart') " locationLink
+                    let l:lines = readfile(fnameescape(l:loc['filename']))
```

Comment:
Can be moved outside of the `if`, just on the off chance we change this line and forget to change it in the else.